packages/vue-app/template/components/nuxt.js

```diff
@@ -40,10 +40,30 @@ export default {
       if (typeof this.nuxtChildKey !== 'undefined' || this.$route.matched.length > 1) {
         return this.nuxtChildKey || compile(this.$route.matched[0].path)(this.$route.params)
       }
+
       const Component = this.$route.matched[0] && this.$route.matched[0].components.default
-      if (Component && Component.options && Component.options.key) {
-        return (typeof Component.options.key === 'function' ? Component.options.key(this.$route) : Component.options.key)
+      if (Component && Component.options) {
+        const { key, watchQuery } = Component.options
+
+        if (key) {
+          return (typeof key === 'function' ? key(this.$route) : key)
+        }
+
+        if (watchQuery) {
+          if (watchQuery.length) {
+            const pickedQuery = {}
+            for (const queryKey of watchQuery) {
+                pickedQuery[queryKey] = this.$route.query[queryKey]
```

Comment:
Please fix indentation to 2 spaces here and other places I noticed it's off.

Reply:
I've fixed wrong indents.